pulp_container/app/registry_api.py

```diff
@@ -336,6 +336,9 @@ def get_dr_push(self, request, path, create=False):
                         distribution = dist_serializer.create(dist_serializer.validated_data)
                 except ValidationError:
                     raise RepositoryInvalid(name=path)
+                except IntegrityError:
+                    # Seems like another process created our stuff already. Retry fetching it.
+                    distribution = models.ContainerDistribution.objects.get(base_path=path)
```

Comment:
`repository = distribution.repository` needs to be added to overcome `variable repository was reference before assignment` error

Reply:
i would love to refactor get_rd_push so it does not look that crazy, maybe not part of this pr.